frontend/src/core/comments/components/AddComment.tsx

```diff
@@ -1,30 +1,55 @@
-/* @flow */
-
 import * as React from 'react';
 import ReactDOM from 'react-dom';
 import { Localized } from '@fluent/react';
-import { Editor, Transforms, Range, createEditor, Text, Node } from 'slate';
+import {
+    BaseEditor,
+    Editor,
+    Element as SlateElement,
+    Transforms,
+    Range,
+    createEditor,
+    Text,
+    Node,
+} from 'slate';
+import type { Descendant } from 'slate';
 import { Slate, Editable, ReactEditor, withReact } from 'slate-react';
 import escapeHtml from 'escape-html';
 
 import './AddComment.css';
 
 import { UserAvatar } from 'core/user';
 
-import type { TextType, MentionType, InitialType } from 'core/api';
 import type { NavigationParams } from 'core/navigation';
 import type { UserState } from 'core/user';
 
-type Props = {|
-    parameters: ?NavigationParams,
-    translation?: ?number,
-    user: UserState,
-    contactPerson?: string,
-    addComment: (string, ?number) => void,
-    resetContactPerson?: () => void,
-|};
+type Props = {
+    parameters: NavigationParams | null | undefined;
+    translation?: number | null | undefined;
+    user: UserState;
+    contactPerson?: string;
+    addComment: (arg0: string, arg1: number | null | undefined) => void;
+    resetContactPerson?: () => void;
+};
+
+type Paragraph = {
+    type: 'paragraph';
+    children: Descendant[];
+};
+
+type Mention = {
```

Comment:
The `Mention` type also includes a `url` which is type `string` (see original typing from `/core/api/types` [HERE](https://github.com/mozilla/pontoon/blob/master/frontend/src/core/api/types.js#L42))

Reply:
Good catch, there's something fishy about TS not complaining about that type being wrong.

Reply:
I force-pushed an update for this.

The `as Paragraph` cast is necessary, it seems. I haven't grokked why, but this way works.